src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -135,7 +136,14 @@ private bool ParseUDFArgs(out HashSet<UDFArg> args)
                 if (TokEat(TokKind.Colon, addError: false) == null)
                 {
                     CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingParamType);
-                    break;
+
+                    // If the result was an error, keep moving cursor until end of named formula expression
+                    while (_curs.TidCur != TokKind.Semicolon && _curs.TidCur != TokKind.Eof)

```

Comment:
If it's missing a param type, could we try to move to the `)` or `;`? maybe the body is still valid? not for this pr, but something to consider, udfs have some other syntax elements that can be used.